_api-reference/ingest-apis/ip2geo.md

```diff
@@ -0,0 +1,247 @@
+
+#### Example: DELETE request
+
+```json
+DELETE /_plugins/geospatial/ip2geo/datasource/my-datasource
+```
+
+#### Example: Response
+
+```json
+{
+  "acknowledged": true
+}
+```
+
+## Creating the processor
+
+Once the IP2Geo data source is created, you can create the `ip2geo` processor. 
+
+#### Example: Create processor request
+
+```json
+{
+   "description":"convert ip to geo",
+   "processors":[
+    {
+        "ip2geo":{
+            "field":"ip",
+            "datasource":"my-datasource"
+        }
+    }
+   ] 
+}
+```
+
+#### Example: Response
+
+```json
+{
+	"acknowledged": true
+}
+```
+
+## Creating the IP2Geo pipeline
+
+The following table lists the `ip2geo` fields options for creating an IP2Geo pipeline.
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| field | yes | - | The field to get the ip address for the geographical lookup. |
+| datasource | yes | - | The data source name to look up geographical information. |
+| properties | no |  All fields in `datasource`. | The field that controls what properties are added to `target_field` from `datasource`. |
+| target_field | no | ip2geo | The field that holds the geographical information looked up from the data source. |
+| database | no | geoip.maps.opensearch.org/v1/geolite2-city/manifest.json | The database filename referring to a database the module ships with or a custom database in the ingest-geoip config directory. |
+| ignore_missing | no | false | If `true` and `field` does not exist, the processor quietly exits without modifying the document. |
+| first_only | no | true | If `true` only first-found GeoIP data is added. This only applies when field contains multiple IPs in an array. |
+
+The following code is an example of using the `ip2geo` processor to add the geographical information to the `ip2geo` field based on the `ip` field.
+
+```json
+PUT /_ingest/pipeline/ip2geo
+{
+   "description":"convert ip to geo",
+   "processors":[
+    {
+        "ip2geo":{
+            "field":"ip",
+            "datasource":"my-datasource"
+        }
+    }
+   ] 
+}
+
+PUT /my-index/_doc/my-id?pipeline=ip2geo
+{
+  "ip": "172.0.0.1"
+}
+
+GET /my-index/_doc/my-id
+{
+   "_index":"my-index",
+   "_id":"my-id",
+   "_version":1,
+   "_seq_no":0,
+   "_primary_term":1,
+   "found":true,
+   "_source":{
+      "my_ip_field":"172.0.0.1",
+      "ip2geo":{
+         "continent_name":"North America",
+         "region_iso_code":"AL",
+         "city_name":"Calera",
+         "country_iso_code":"US",
+         "country_name":"United States",
+         "region_name":"Alabama",
+         "location":"33.1063,-86.7583",
+         "time_zone":"America/Chicago"
+      }
+   }
+}
+```
+
+## Node settings
+
+The IP2Geo data source and `ip2geo` processor node settings are listed in the following table.
+
+| IP2Geo data source | Description | Setting |
```

Comment:
Would be better not to separate IP2Geo datasource setting and processor setting.
```suggestion
| Key | Description | Default |
```